client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx

```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
+      apiCalls.getJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+      apiCalls.postRound({ 'contest-id': 67 }),
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      renderView()
+      const newSampleSize = await screen.findByText(
+        '67 samples (70% chance of reaching risk limit and completing the audit in one round)'
       )
-    )
-    render(
-      <Router>
-        <Review locked={false} prevStage={prevStage} refresh={jest.fn()} />
-      </Router>
-    )
-    const newSampleSize = await screen.findByText(
-      '67 samples (70% chance of reaching risk limit and completing the audit in one round)'
-    )
-    fireEvent.click(newSampleSize, { bubbles: true })
-    const launchButton = screen.getByText('Launch Audit')
-    fireEvent.click(launchButton, { bubbles: true })
-    await screen.findByText('Are you sure you want to launch the audit?')
-    const confirmLaunchButton = screen.getAllByText('Launch Audit')[1]
-    fireEvent.click(confirmLaunchButton, { bubbles: true })
-    await waitFor(() => {
-      expect(apiMock).toHaveBeenCalledTimes(5)
-      expect(apiMock.mock.calls[4][1]).toMatchObject({
-        body: JSON.stringify({
-          sampleSize: 67,
-          roundNum: 1,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        method: 'POST',
-      })
+      userEvent.click(newSampleSize)
+      const launchButton = await screen.findByText('Launch Audit')
+      userEvent.click(launchButton)
+      await screen.findByText('Are you sure you want to launch the audit?')
+      const confirmLaunchButton = screen.getAllByText('Launch Audit')[1]
+      userEvent.click(confirmLaunchButton)
+      await waitFor(() => expect(refreshMock).toHaveBeenCalled())
     })
   })
 
   it('accepts custom sample size', async () => {
-    apiMock.mockImplementation(
-      generateApiMock(
-        sampleSizeMock,
-        contestMocks.filledTargetedWithJurisdictionId,
-        { jurisdictions: jurisdictionMocks.allManifests }
+    auditSettingsMock.mockReturnValue(settingsMock.full)
+    const expectedCalls = [
+      apiCalls.getJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+      apiCalls.postRound({ 'contest-id': 5 }),
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      renderView()
+      const newSampleSize = await screen.findByText(
+        'Enter your own sample size (not recommended)'
+      )
+      userEvent.click(newSampleSize)
+      const customSampleSizeInput = await screen.findByRole('textbox')
+      fireEvent.change(customSampleSizeInput, { target: { value: '40' } }) // userEvent has a problem with this field due to the lack of an explicit value field: https://github.com/testing-library/user-event/issues/356
```

Comment:
that's annoying. why doesn't it have a value field? isn't it a regular `input`?

Reply:
@jonahkagan It's wrapped a bunch of times, but ultimately it's the [input group element from Blueprint](https://blueprintjs.com/docs/#core/components/text-inputs). And from examining how they handle things, it looks like the fields don't have a value attribute until you start entering data into them. Which would explain why userEvent is getting confused.